easybuild/framework/easyconfig/easyconfig.py

```diff
@@ -1226,6 +1226,27 @@ def get_parsed_multi_deps(self):
 
         return multi_deps
 
+    def find_dep_version_match(self, dep):
+        """Identify the correct version for this system from the choices provided. This overwrites dep['version']"""
+        if isinstance(dep['version'], string_type):
+            self.log.debug("Version is already a string ('%s'), OK", dep['version'])
+            return
+        elif dep['version'] is None:
+            self.log.debug("Version is None, OK", dep)
+        elif isinstance(dep['version'], dict):
+            # figure out matches based on dict keys (after splitting on '=')
+            my_arch_key = 'arch=%s' % get_cpu_architecture()
+            has_arch_keys = [x for x in dep['version'].keys() if x.startswith('arch=')]
+            if has_arch_keys:
```

Comment:
@edmondac I think this should be made a bit more strict...

Right now, only `arch=` keys are supported, so we should make sure nothing else is used (and if so, an appropriate error should be raised).